### ISSUE TITLE
Standardized message when status is DONE_FAILED

### ISSUE THREAD

**Issue by joeseos** [on-api/on-api#36]
I would like the messages when the Order API return status DONE_FAILED to be standardized, I.e. for example, when the port is blocked, then the message could be "port is blocked", then the SP can build logic to manage these errors in a better way.

Do you agree and do you have any suggestions on standardized messages?

**Comment by thoho** [comment]
I agree wholly that it is a good idea to be able to process a "failed" message programmatically but I have opinions against a general text "message" field being parsed and being dependent on either a long string comparison or having certain magical words in human readable text to which attach strict semantics.

In my opinion, "message" fields should have loose syntax requirements and by definition always be intended for human interpretation.

Instead I would suggest complementing the message field with a new field that is intended for automatic processing and with a set of much shorter values to indicate the reason for the failure.

**Comment by DanielMaintrac** [comment]
I agree with @thoho, an "enum" for errors is much better than long string comparisons. Perhaps we could generate an UUID for certain unrecoverable errors that can be used to reference errors that needs to be handled manually.

**Comment by joeseos** [comment]
Maybe it's a good idea to expand the 'state', for example "DONE_FAILED_BLOCKED_PORT"?

**Comment by joakimbeng** [comment]
I agree with @thoho on this, maybe we can add a field like `code` to the response that contains the standard error code and the message can then still be human readable as it is now

**Comment by teemufrosen** [comment]
Working group meeting: Needs further discussion, will be handled in future version.

**Comment by junetaw** [comment]
This could also connect with Forced port takeover and "Porting structure" that is a major problem.

**Comment by teemufrosen** [comment]
Working group meeting: Still needs further discussion, will be handled in a subgroup.

**Comment by junetaw** [comment]
Should be renamed to "Failed order activations and takeover"? Or something broader than currently? If scope is changed...
